src/Apps/Order/Routes/Shipping2/FulfillmentDetails.tsx

```diff
@@ -66,16 +74,150 @@ export const FulfillmentDetails: FC<FulfillmentDetailsProps> = props => {
     setVerifyAddressNow(false)
   }
 
+  // Once the user sees the address form, they should always see it.
+  const [forceNewAddressFormMode, setForceNewAddressFormMode] = useState(
+    !hasSavedAddresses
+  )
+
+  useEffect(() => {
+    if (!forceNewAddressFormMode && !hasSavedAddresses) {
+      setForceNewAddressFormMode(true)
+    }
+  }, [forceNewAddressFormMode, hasSavedAddresses])
```

Comment:
Moved up from the form layout because we only perform address updates _after_ submit when we are in 'new address form' (not address modal) mode.